kafka-client/src/main/java/dev/responsive/kafka/api/ResponsiveStores.java

```diff
@@ -0,0 +1,235 @@
+package dev.responsive.kafka.api;
+
+import dev.responsive.kafka.store.ResponsiveMaterialized;
+import dev.responsive.kafka.store.ResponsiveStoreBuilder;
+import dev.responsive.utils.StoreUtil;
+import java.time.Duration;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.kstream.Materialized;
+import org.apache.kafka.streams.state.KeyValueBytesStoreSupplier;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+import org.apache.kafka.streams.state.Stores;
+import org.apache.kafka.streams.state.TimestampedKeyValueStore;
+import org.apache.kafka.streams.state.TimestampedWindowStore;
+import org.apache.kafka.streams.state.WindowBytesStoreSupplier;
+import org.apache.kafka.streams.state.WindowStore;
+
+/**
+ * A factory for creating Kafka Streams state stores on top of a Responsive storage backend.
+ * Use these when building your {@link org.apache.kafka.streams.Topology} to easily swap in
+ * Responsive stores wherever state is used.
+ * <p>
+ * See {@link org.apache.kafka.streams.state.Stores} for instructions on how to plug in custom
+ * state stores and configure them.
+ */
+public final class ResponsiveStores {
+
+  /**
+   * See for example {@link Stores#inMemoryKeyValueStore(String)}
+   *
+   * @param name the store name
+   * @return a supplier for a key-value store with the given options
+   *         that uses Responsive's storage for its backend
+   */
+  public static KeyValueBytesStoreSupplier keyValueStore(final String name) {
```

Comment:
also yes... I am aware that it's technically a "persistent" store since we return `false` for that, but for all intents and purposes (and as far as the user cares) it is persistent.

Reply:
But this is a lie... semantically we're closer to the in-memory variants of the store suppliers (related to https://github.com/responsivedev/responsive-pub/pull/53#discussion_r1251334656)

(Also why should they get a compile error for IM stores?)

Reply:
If the goal is to make migration stupid simple, simply doing a find-and-replace to change the method name seems well within that

Not to mention, and this is going to be part of the larger discussion so I won't get into it too much, but the (admittedly clever) `s/Stores/ResponsiveStores/g` swap really will only help what is likely to be a very small minority of users, who are already plugging in custom stores everywhere they possibly have state...

Reply:
heh, I like the elegance of a true `s//g` but I'll concede here - it's probably fine to keep it as is. I won't block the PR on it.

>  will only help what is likely to be a very small minority of users

I guess my sample size is 1 ;) but for the others I think the eventual path is to support `dsl.default.store` to be a custom store implementation. Also wouldn't this apply to anyone using the Processor API?

Reply:
> the eventual path is to support `dsl.default.store` to be a custom store implementation.

Yeah I 100% agree this should be our API #goalz -- or something like it that also applies to the PAPI.

> Also wouldn't this apply to anyone using the Processor API?

Strict PAPI users yes, but in my experience most users who touch the PAPI still do it via transformers -- ie they use a mix of DSL and PAPI. This is what Metronome does IIRC. And we don't want them to miss plugging in our stores to some random stateful DSL operator elsewhere in their topology. 

I have no idea what the overall userbase breaks down into w.r.t DSL vs PAPI vs mix, but if I had to guess I'd put money on DSL or mix heavily outweighing PAPI-only. As to how many of the "mix" category actually has state in the DSL or the PAPI or both, well, I wouldn't put money on that bet because I really have no idea 😄